lnwallet/channel_test.go

```diff
@@ -8987,617 +9048,6 @@ type heights struct {
-				localAdd:     height,
-				localRemove:  0,
-				remoteAdd:    height,
-				remoteRemove: 0,
-			},
-			expectedHeights: heights{
-				localAdd:     height,
-				localRemove:  0,
-				remoteAdd:    height,
-				remoteRemove: 0,
-			},
-			whoseCommitChain: lntypes.Remote,
-			mutate:           false,
-			expectedFee:      feePerKw,
-		},
-		{
-			// Local add height is non-zero, so the update has
-			// already been applied; we do not expect fee to
-			// change or any mutations to be applied.
-			name: "non-zero local height, mutation not applied",
-			startHeights: heights{
-				localAdd:     height,
-				localRemove:  0,
-				remoteAdd:    0,
-				remoteRemove: height,
-			},
-			expectedHeights: heights{
-				localAdd:     height,
-				localRemove:  0,
-				remoteAdd:    0,
-				remoteRemove: height,
-			},
-			whoseCommitChain: lntypes.Local,
-			mutate:           true,
-			expectedFee:      feePerKw,
-		},
-		{
-			// Local add is zero and we are looking at our local
-			// chain, so the update has not been applied yet. We
-			// expect the local add and remote heights to be
-			// mutated.
-			name: "zero height, fee changed, mutation applied",
-			startHeights: heights{
-				localAdd:     0,
-				localRemove:  0,
-				remoteAdd:    0,
-				remoteRemove: 0,
-			},
-			expectedHeights: heights{
-				localAdd:     nextHeight,
-				localRemove:  nextHeight,
-				remoteAdd:    0,
-				remoteRemove: 0,
-			},
-			whoseCommitChain: lntypes.Local,
-			mutate:           true,
-			expectedFee:      ourFeeUpdatePerSat,
-		},
-	}
-
-	for _, test := range tests {
-		test := test
-
-		t.Run(test.name, func(t *testing.T) {
-			// Create a fee update with add and remove heights as
-			// set in the test.
-			heights := test.startHeights
-			update := &paymentDescriptor{
-				Amount:                   ourFeeUpdateAmt,
-				addCommitHeightRemote:    heights.remoteAdd,
-				addCommitHeightLocal:     heights.localAdd,
-				removeCommitHeightRemote: heights.remoteRemove,
-				removeCommitHeightLocal:  heights.localRemove,
-				EntryType:                FeeUpdate,
-			}
-
-			view := &HtlcView{
-				FeePerKw: chainfee.SatPerKWeight(feePerKw),
-			}
-			processFeeUpdate(
-				update, nextHeight, test.whoseCommitChain,
-				test.mutate, view,
-			)
-
-			if view.FeePerKw != test.expectedFee {
-				t.Fatalf("expected fee: %v, got: %v",
-					test.expectedFee, feePerKw)
-			}
-
-			checkHeights(t, update, test.expectedHeights)
-		})
-	}
-}
-
-func checkHeights(t *testing.T, update *paymentDescriptor, expected heights) {
-	updateHeights := heights{
-		localAdd:     update.addCommitHeightLocal,
-		localRemove:  update.removeCommitHeightLocal,
-		remoteAdd:    update.addCommitHeightRemote,
-		remoteRemove: update.removeCommitHeightRemote,
-	}
-
-	if !reflect.DeepEqual(updateHeights, expected) {
-		t.Fatalf("expected: %v, got: %v", expected, updateHeights)
-	}
-}
-
-// TestProcessAddRemoveEntry tests the updating of our and their balances when
```

Comment:
I understand that we do not need to check the mutate state anymore but where do we now test that the balances are updated correctly for the different payment descriptors ?

Reply:
Moreover the mutation is now pulled out into an upper level, are there unit-tests for it ? Cannot see a unit-test for `computeView` which now does the mutating.

Reply:
as in the lifetime payment flows? I can try to design a test around that but I don't think the lifetime payment flow stuff is mission critical. With regards to the state updates of the payment descriptor commit heights, that is already implicitly tested by the itests during any payment that gets made, I think.

Reply:
ok normally I tend to favour having a complete unit-test suite rather than relying on itests, but given the fact that computeview had no unit-test beforehand having an unit-test for `computeView` would be nice but it's a non-blocking comment.